packages/context/src/provider.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Context} from './context';
+import {BoundValue, ValueOrPromise} from './binding';
+import {Constructor, instantiateClass} from './resolver';
+
+/**
+ * Provider of a value
+ */
+export interface Provider<T> {
+  value(ctx?: Context): ValueOrPromise<T>;
```

Comment:
> I'm also fine to even remove ctx (i.e., value()) as the ctx or fine-grained dependencies can be always injected for the class.

I like this even better, as it nudges people to inject dependencies via `inject` and don't access `context` directly 👍